spec/mem/tree_borrows/memory.md

```diff
@@ -85,19 +101,48 @@ impl<T: Target> TreeBorrowsMemory<T> {
             return ret(ptr);
         };
 
+        let freeze_info = ptr_type.safe_pointee().map(|pointee_info| pointee_info.freeze);
+        let pointee_nonfreeze_bytes = freeze_info.map(|strategy| {
+            match strategy {
+                UnsafeCellStrategy::Sized { bytes } => bytes,
+                // TODO: Implement for unsized cases
+                _ => List::new(),
+            }
+        });
+        let is_freeze = freeze_info.map_or(true, |strategy| strategy.is_freeze());
+        let protected = new_perm.protected;
+
         let child_path = self.mem.allocations.mutate_at(alloc_id.0, |allocation| {
+            let size = allocation.size();
+
+            let location_states: List<LocationState> = if pointee_nonfreeze_bytes.is_some() {
+                (Int::ZERO..size.bytes()).map(|i| {
+                    let i = Size::from_bytes(i).unwrap();
+                    // Check if `i` is included in any of the ranges. O(size * |pointee_nonfreeze_bytes|)
+                    // TODO: More performant way to do this?
```

Comment:
You might want to create a `RangeMap`-like wrapper type for these lists here, where there is an efficient method for iterating the bytes one-by-one but in such a way that this has `O(n+m)` where `n` is the number of bytes and `m` is the number of entries in the list. This should presumably be "abstracted" into its own function.